### PR TITLE
Added mark as success for mass action and single action

### PR DESCRIPTION
**Pull request by jnmrclmbsjse**
This PR includes:
1. Added mark as success buttton for action column
2. Changed retry link to button
3. Added resolve action dropdown which has two options `Retry` and `Mark as success`
4. Fixed some code alignment
image

### CODE REVIEW

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php [hunk 1]
```diff
@@ -31,36 +33,50 @@ public function resolveTaskAction()
...
         }
-        wp_redirect(remove_query_arg(['selected', 'action']));
+        $this->clearArgs();
     }
 
     public function resolveTasksAction()
     {
-        if (array_key_exists('selected', $_POST)) {
+        if (array_key_exists('selected', $_POST) && array_key_exists('resolveAction', $_POST)) {
+            $resolveAction = $_POST['resolveAction'];
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php:46]: I think it's just a row action

```suggestion
            $rowAction = $_POST['rowAction'];
```

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php [hunk 2]
```diff
@@ -158,27 +174,41 @@ public function renderTaskActions($value, $task)
             case TaskHandler::STATUS_NEW:
             case TaskHandler::STATUS_SUCCESS:
                 echo <<<HTML
-<div class="storekeeper-status">
-    <span class="storekeeper-status-success"></span>
-</div>
-HTML;
+                    <div class="storekeeper-status">
+                        <span class="storekeeper-status-success"></span></center>
```
[the remark this answers is not in the document]
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php]: Removed unnecessary center tag.

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php [hunk 3]
```diff
@@ -10,6 +10,8 @@ class TaskLogsTab extends AbstractLogsTab
 {
     const ACTION_RESOLVE = 'action-resolve';
     const ACTION_MASS_RESOLVE = 'action-mass-resolve';
+    const RESOLVE_RETRY_ACTION = 'retry';
+    const RESOLVE_MARK_ACTION = 'mark';
```
[the remark this answers is not in the document]
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php]: Renamed variables as suggestion stated.

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php (hunk 1 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php]: Renamed keys and variables as suggestion stated.

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php [hunk 4]
```diff
@@ -31,36 +33,50 @@ public function resolveTaskAction()
...
-                ]
+                ],
+                $this->getRowAction(),
             );
         }
-        wp_redirect(remove_query_arg(['selected', 'action']));
+        $this->clearArgs();
     }
 
     public function resolveTasksAction()
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php:43]:
```suggestion
    public function rowAction()
```

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php [hunk 5]
```diff
@@ -31,36 +33,50 @@ public function resolveTaskAction()
...
             $selected = $this->sanitizeIntArray($_POST['selected']);
-            $this->resolveTasks($selected);
+            $this->resolveTasks($selected, $rowAction);
         }
-        wp_redirect(remove_query_arg(['selected', 'action']));
+        $this->clearArgs();
     }
 
-    private function resolveTasks(array $taskIds)
+    private function resolveTasks(array $taskIds, string $rowAction)
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php:53]:
```suggestion
    private function doRowAction(array $taskIds, string $rowAction)
```

**Review by ikulis — COMMENTED** [review]
missed 2 spots on rename

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php (hunk 5 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php]: Changed to `doRowAction`

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php [hunk 6]
```diff
@@ -263,20 +293,32 @@ private function renderTaskStatusFilter()
...
+        $optionHtml = "<option value=''>$optionLabel</option>";
+        $rowActions = [
+            self::RETRY_ACTION => esc_html__('Retry', I18N::DOMAIN),
+            self::MARK_ACTION => esc_html__('Mark as success', I18N::DOMAIN),
+        ];
+        foreach ($rowActions as $value => $label) {
+            $selected = $currentAction === $value ? 'selected' : '';
+            $optionHtml .= "<option value='$value' $selected>$label</option>";
+        }
+        $resolve = esc_html__('Apply', I18N::DOMAIN);
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php:317]: missed it

Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/TaskLogsTab.php]: Changed to `$apply`

**Pull request merged**